deeplake/core/vectorstore/vector_search/utils.py

```diff
@@ -19,6 +21,19 @@ def parse_tensor_return(tensor):
     return tensor.data(aslist=True)["value"]
 
 
+def parse_exec_option(dataset, exec_option):
+    if exec_option == "auto":
+        if isinstance(dataset, DeepLakeCloudDataset):
```

Comment:
What about `gcs`, `s3`, `azure`, do we want for all of them by default use `compute_engine`?  If so, we will need to do a quick follow up PR to LangChain to explicitly use python be default, as users might start getting problems. With this implementation it will try to use `python` for these storage providers. We need to make that consistent.

Reply:
See comment above.